### PR TITLE
Don't display loading indicator over channels from cache

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### Description

This PR prevent channel list loading indicator from displaying over channel list received from cache. This is just a temp solution. We need to completely get rid of sealed classes in our view models and controllers:

```
data class State(
    val contentState: ContentState,
    val channels: List<Channel>,
    val loadingMore: Boolean,
    val hasMore: Boolean
)

enum class ContentState {
    CONTENT, LOADING, ERROR
}
```

### Checklist

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [X] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [X] Comparison screenshots added for visual changes
- [X] Reviewers added

### Before
https://user-images.githubusercontent.com/9600921/102942457-6d209f80-44c6-11eb-9646-aad1e06f9334.mov

### After
https://user-images.githubusercontent.com/9600921/102942452-6a25af00-44c6-11eb-8ffc-8bef3644bec9.mov

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt [hunk 1]
```diff
@@ -26,16 +26,19 @@ public class ChannelListView @JvmOverloads constructor(
     defStyle: Int = 0
 ) : RecyclerView(context, attrs, defStyle) {
 
-    private var endReachedListener: EndReachedListener? = null
+    private val channelListItemAdapter: ChannelListItemAdapter
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt:29]: Why do we need to store one extra field? We can reuse just `adapter` for the same purpose, can't we?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt [hunk 2]
```diff
@@ -124,6 +127,10 @@ public class ChannelListView @JvmOverloads constructor(
         requireAdapter().submitList(channels)
     }
 
+    public fun hasChannels(): Boolean {
+        return channelListItemAdapter.itemCount > 0
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt:131]: `adapter?.itemCount ?: 0 > 0`
Isn't the same?

**Review by ogkuzmin — CHANGES_REQUESTED** [review]

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt:29]: There's a very minor difference, `BaseAdapter` not only provides items that were set to the adapter, but also pending items in async differ.

Comment (bychkovdmitry) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt:29]: So I felt, even though there is a very small chance chance that we invoke the method after we set channels but before async diff completes, it seems like the right thing to do.

**Review by ogkuzmin — APPROVED** [review]

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt:29]: Ok, seems that I'm wrong here. Fixing it...

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**